harvester/records/formats/marc.py

```diff
@@ -20,70 +41,197 @@ class MARC(MarcalyxSourceRecord):
+        if bbox_data is None:
+            return None
+
+        return (
+            f"ENVELOPE({bbox_data['w']}, "
+            f"{bbox_data['e']}, "
+            f"{bbox_data['n']}, "
+            f"{bbox_data['s']})"
+        )
 
     def _locn_geometry(self) -> str | None:
-        raise NotImplementedError
+        """Field method: locn_geometry
+
+        Some MARC records have a set of coordinates in 034 where the E/W and N/S
+        coordinates are the same, effectively defining a WKT POINT.  If this is true
+        across even multiple 034 tags, return a WKT POINT for this field.
+        """
+        bbox_data = self._get_largest_bounding_box()
+        if bbox_data is None:
+            return None
+
+        if bbox_data["w"] == bbox_data["e"] and bbox_data["n"] == bbox_data["s"]:
+            return f"POINT({bbox_data['w']}, {bbox_data['n']})"
+
+        return self._dcat_bbox()
 
     ##########################
     # Optional Field Methods
     ##########################
 
     def _dct_description_sm(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _dcat_keyword_sm(self) -> list[str]:
         """New field in Aardvark: no mapping from GBL1 to dcat_keyword_sm."""
-        raise NotImplementedError
+        return []
 
     def _dct_alternative_sm(self) -> list[str]:
         """New field in Aardvark: no mapping from GBL1 to dct_alternative_sm."""
-        raise NotImplementedError
+        return []
 
     def _dct_creator_sm(self) -> list[str] | None:
-        raise NotImplementedError
+        return None
 
     def _dct_format_s(self) -> str | None:
-        raise NotImplementedError
+        return None
 
     def _dct_issued_s(self) -> str | None:
-        raise NotImplementedError
+        return None
 
     def _dct_identifier_sm(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _dct_language_sm(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _dct_publisher_sm(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _dct_rights_sm(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _dct_spatial_sm(self) -> list[str] | None:
-        raise NotImplementedError
+        return None
 
     def _dct_subject_sm(self) -> list[str] | None:
-        raise NotImplementedError
+        return None
 
     def _dct_temporal_sm(self) -> list[str] | None:
-        raise NotImplementedError
+        return None
 
     def _gbl_dateRange_drsim(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _gbl_resourceType_sm(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _gbl_indexYear_im(self) -> list[int]:
-        raise NotImplementedError
+        return []
+
+    ##########################
+    # Helpers
+    ##########################
+
+    def _get_largest_bounding_box(
```

Comment:
I think this would be easier to parse if it were broken up

Reply:
Agreed.  I had considered it, but was worried about too many helper methods.  But, better that than a method that is hard to reason about.  Update forthcoming.

Reply:
@ehanson8 - here is an updated version with this logic broken out a bit: https://github.com/MITLibraries/geo-harvester/blob/GDT-227-transform-alma-marc-required-fields/harvester/records/formats/marc.py#L155-L201.

I struggled with this one.  Sometimes with more imperative code, you can get a nice flow of logging and returning early if something isn't right.  But, I realized the logging was probably overkill here, and with that removed, I think it flows better.

I think there is a balance between logging that alerts to unusual behavior vs that which can be understood just through analysis and working through a particular use case.  My thinking is that less logging here -- e.g. missing or invalid subfields -- would just be the work of analyzing why a particular record may have failed.  Given we don't have `DEBUG` logging on when deployed, it wouldn't surface that way anyways.